gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/ContextAwareMetricFactory.java

```diff
@@ -107,6 +120,23 @@ public boolean isInstance(Metric metric) {
     }
   }
 
+  /**
+   * An implementation of {@link ContextAwareMetricFactory} using {@link com.codahale.metrics.SlidingTimeWindowReservoir}
+   * for {@link ContextAwareHistogram} creation.
+   */
+  public static class SlidingTimeWindowContextAwareHistogramFactory extends ContextAwareHistogramFactory {
```

Comment:
Don't fully understand what do you mean by method implementations. The factory method a common method instead of binding with a specific metrics type, right?